test.js

```diff
@@ -0,0 +1,227 @@
+const subject = require('./');
+const assume = require('assume');
+const fs = require('fs');
+
+describe('verify', () => {
+  let pubkey;
+  let document;
+  let pkcs7;
+  let pkcs7_with_header;
+
+  beforeEach(() => {
+    pubkey = fs.readFileSync('./test-files/rsa2048-pubkey');
+    document = fs.readFileSync('./test-files/document');
+    pkcs7 = fs.readFileSync('./test-files/rsa2048');
+    pkcs7_with_header = fs.readFileSync('./test-files/rsa2048-with-header');
+
+  });
+
+  it('should be a function', () => {
+    assume(subject).is.function();
+  });
+
+  describe('with missing values', () => {
+    it('should throw when no values are provided', () => {
+      assume(() => {
+        subject();
+      }).throws(/must be provided$/);
+    });
+    it('should throw when pubkey isnt provided', () => {
+      assume(() => {
+        subject(undefined, 'y', 'z');
+      }).throws(/^pubkey must be provided$/);
+    });
+    it('should throw when document isnt provided', () => {
+      assume(() => {
+        subject('x', undefined, 'z');
+      }).throws(/^document must be provided$/);
+    });
+    it('should throw when pkcs7 isnt provided', () => {
+      assume(() => {
+        subject('x', 'y', undefined);
+      }).throws(/^pkcs7 signature must be provided$/);
+    });
+  });
+
+  describe('with bad values', () => {
+    it('should return false with empty Buffer values', () => {
```

Comment:
nit: looks like the test descriptions got out of sync and we can replace "return false" with "throw error" in this describe block

Reply:
you're right, thanks!